### ISSUE TITLE
Bug: PasswordField ShowPassword button context not announced to AT

### ISSUE THREAD

**Issue by hbuchel** [aws-amplify/amplify-ui#1810]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Any

### Please describe your bug.

Currently, the ShowPasswordButton does not announce its change in state when toggled. Additionally, the aria-label provided which switches back and forth between "Hide label" and "Show label" has somewhat confusing context.

In attached video, notice that when the button is toggled on and off, nothing is announced.

https://user-images.githubusercontent.com/376920/166312889-7b07ed39-a79e-4040-a8a6-f58571036ae6.mp4

.

### What's the expected behaviour?

The change in state of the password input should be announced. When toggled "on", it should announce "Password is shown." When toggled "off", it should announce "Password is hidden." Additionally, we should consider using a switch role for the button to include the context of it turning on and off, and keep a consistent label (as you would for a checkbox, the label stays the same). We have a spike completed to show acceptance criteria for this bug and the recommended fix.

### Help us reproduce the bug!

Any use of the PasswordField component. This can be demonstrated in the docs or in an external application instance using either the primitive or the Authenticator component.

### Code Snippet

```typescript
// Put your code below this line.

```
